internal/configprovider/config_server.go

```diff
@@ -0,0 +1,165 @@
+// Copyright Splunk, Inc.
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configprovider
+
+import (
+	"net"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/spf13/cast"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	defaultConfigServerEndpoint   = "localhost:55555"
+	defaultConfigServerPortEnvVar = "SPLUNK_CONFIG_SERVER_PORT"
+)
+
+type configServer struct {
+	logger    *zap.Logger
+	initial   map[string]interface{}
+	effective map[string]interface{}
+	server    *http.Server
+	doneCh    chan struct{}
+}
+
+func newConfigServer(logger *zap.Logger, initial, effective map[string]interface{}) *configServer {
+	return &configServer{
+		logger:    logger,
+		initial:   initial,
+		effective: effective,
+	}
+}
+
+func (cs *configServer) start() error {
+	endpoint := defaultConfigServerEndpoint
+	if portOverride, ok := os.LookupEnv(defaultConfigServerPortEnvVar); ok {
+		if portOverride == "" {
```

Comment:
I am making the default to expose the config on the default port let me know if this should be disabled by default.

Reply:
Even w/ redacted fields, potentially exposing sensitive info by default doesn't seem ideal imo.  What about adding a flag like `SPLUNK_DEBUG_CONFIG_SERVER` that defaults to false if this can't easily be integrated w/ the config providers upstream?

Reply:
This will require a re-start but it seems fair to do it at least for initial implementation.